### PR TITLE
feat: Enable HotReload

### PR DESCRIPTION
**Pull request by nickrandolph**
## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Adds packages and code to enable Hot Reload

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Templates/Uno.Templates.csproj [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 	<PropertyGroup>
 		<UnoExtensionsVersion Condition="'$(UnoExtensionsVersion)' == ''">3.0.0-dev.2335</UnoExtensionsVersion>
-		<UnoVersion Condition="'$(UnoVersion)' == ''">5.0.0-dev.3597</UnoVersion>
+		<UnoVersion Condition="'$(UnoVersion)' == ''">5.1.0-dev.52</UnoVersion>
```
Comment (nickrandolph) [inline src/Uno.Templates/Uno.Templates.csproj:21]: This needs to be a 5.0.x version before this is merged

File: src/Uno.Templates/reinstall.ps1 [hunk 2]
```diff
@@ -6,7 +6,7 @@ param(
     [string]$ExtensionsVersion = "3.0.0-dev.2335",
 
     # Version of published Uno.WinUI packages
-    [string]$UnoVersion = "5.0.0-dev.3597"
+    [string]$UnoVersion = "5.1.0-dev.52"
```
Comment (nickrandolph) [inline src/Uno.Templates/reinstall.ps1:9]: This needs to be a 5.0.x version

**Comment by nickrandolph** [comment]
Blocked waiting on https://github.com/unoplatform/uno/pull/14142

**Review by dansiegel — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.0

**Pull request merged**